File: app/src/main/java/fr/nihilus/music/database/AppDatabase.java
package fr.nihilus.music.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;

@Database(entities = {MusicInfo.class, Playlist.class, PlaylistTrack.class}, version = 1)
@TypeConverters({Converters.class})
public abstract class AppDatabase extends RoomDatabase {
    public static final String NAME = "music.db";

    public abstract MusicInfoDao musicInfoDao();
    public abstract PlaylistDao playlistDao();

    // TODO Setup when opened for the first time
}

File: app/src/main/java/fr/nihilus/music/database/AppDatabase.kt
/*
 * Copyright 2018 Thibault Seisel
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package fr.nihilus.music.database

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.TypeConverters

@Database(entities = [MusicInfo::class, Playlist::class, PlaylistTrack::class], version = 1)
@TypeConverters(Converters::class)
abstract class AppDatabase : RoomDatabase() {

    abstract val musicInfoDao: MusicInfoDao
    abstract val playlistDao: PlaylistDao

    companion object {
        const val NAME = "music.db"
    }
}
